pkg/tracing/tracing.go

```diff
@@ -500,10 +502,13 @@ func (t *Tracer) NewSpan(ctx context.Context, name string) *Span {
 }
 
 // NewSpanWithStart creates a span with specify start time.
-func (t *Tracer) NewSpanWithStart(ctx context.Context, name string, startAt time.Time) *Span {
+func (t *Tracer) NewSpanWithStart(ctx context.Context, name string, startAt time.Time, req *http.Request) *Span {
```

Comment:
according to the design, tracing is protocol independent (though there's a function `InjectHTTP`), this change will make it HTTP-specific.